builder/builder_test.go

```diff
@@ -0,0 +1,95 @@
+package builder
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_Builder(t *testing.T) {
+	t.Parallel()
+
+	for _, test := range []struct { //nolint:govet
+		name, expected string
+		b              *Builder
+	}{
+		{
+			"text",
+			"Hello, World!",
+			New().Text("Hello, World!"),
+		},
+		{
+			"simple message, first character is dot",
+			"\\.text",
+			New().Text(".text"),
+		},
+		{
+			"text, with special chars",
+			"Hello\\\\, \\{World\\}!",
+			New().Text("Hello\\, {World}!"),
+		},
+		{
+			"text with literal",
+			"Hello, { |World| }!",
+			New().
+				Text("Hello, ").
+				Expr(Expr().Literal("World")).
+				Text("!"),
+		},
+		{
+			"text with expr",
+			"Hello, { $world :upper limit = 2 }!",
+			New().
+				Text("Hello, ").
+				Expr(Expr().Var("$world").Func(":upper", Option("limit", "2"))).
+				Text("!"),
+		},
+		{
+			"text with markup-like function",
+			"Hello { +link } World { -link }",
+			New().
+				Text("Hello ").
+				Expr(Expr().Func("+link")).
+				Text(" World ").
+				Expr(Expr().Func("-link")),
+		},
+		{
+			"local",
+			".local $hostName = { $host }\n",
```

Comment:
Isn't the idea of a builder to construct, a valid mf2 syntax message, and error if something is missing/incorrect?

Because this is not a valid complex message

Reply:
ok, will look into this one. Currently, we are only focusing on the happy path.

Reply:
@VladislavsPerkanuks please state what is not correct! :)

According to my knowledge and ABNF spec, it is correct.

Reply:
```abnf
```